src/rsa_api/rsa_api.py

```diff
@@ -1379,6 +1385,50 @@ def IQSTREAM_GetEnable(self) -> bool:
         self.err_check(self.rsa.IQSTREAM_GetEnable(byref(enabled)))
         return enabled.value
 
+    def IQSTREAM_GetIQData(
+        self, dtype: str, buffer_size: int
+    ) -> Tuple[np.ndarray, _IQStreamIQInfo]:
+        """
+        Retrieve interleaved IQ data generated by IQ Stream processing.
+
+        Parameters
+        ----------
+        dtype : str
+            IQSTREAM data type, must be one of: "SINGLE" "INT32"
+            "INT16" or "SINGLE_SCALE_INT32"
+        buffer_size : int
+            Configured size of the IQSTREAM buffer, as returned
+            by IQSTREAM_GetIQDataBufferSize().
+
+        Returns
+        -------
+        np.ndarray
+            Interleaved IQ data retrieved from the buffer.
+        int
+            Length of the IQ block retrieved, in samples.
+        _IQStreamIQInfo
+            IQ Stream info structure, containing the acquisition
+            status. Note that the timestamp field of this structure
+            seems to give wildly incorrect results.
+
+        Raises
+        ------
+        ValueError
+            If `dtype` is not a valid IQSTREAM data type.
+        """
+        np_dtype = _NUMPY_IQS_OUT_DTYPE[_IQS_OUT_DTYPE.index(dtype)]
+        iqdata = np.empty(2 * buffer_size, np_dtype)
+        c_iqdata = np.ctypeslib.as_ctypes(iqdata)
+        iqlen = c_int32()
+        iqinfo = _IQStreamIQInfo()
+
+        self.err_check(
+            self.rsa.IQSTREAM_GetIQData(byref(c_iqdata), byref(iqlen), byref(iqinfo))
+        )
+
+        del c_iqdata
+        return iqdata, iqlen.value, iqinfo
```

Comment:
It looks like method return type needs to be updated to include iqlen

Reply:
Good catch- fixed in [46002b1](https://github.com/NTIA/tekrsa-api-wrap/pull/13/commits/46002b1b3c5ffc54f128cf349e8c8002d14f7f98)